### ISSUE TITLE
Current vnode `dom` property is undefined in `onbeforeupdate`

### ISSUE THREAD

**Issue by martinboulais** [MithrilJS/mithril.js#2887]
**Mithril.js version: 2.2.2**

**Browser and OS: Firefox 118.0.2 (64-bit) on Ubuntu 22.04.3 LTS**

## Code
[flems demo](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvAHgEIARAeQGEAVAJoAFAKIACEmQB8AHTQtJUMVAxoA5gF4ZIJttnzCMDABN9Y82JY5iGMdUIYATnBjEtIAKp8AYgFoAHHpyFpbEEMSwUnzwxCwA9GERMPrxhiYpAEa0xgCeKXFZufmK+pQgLrDUYfQIiCAAbIj1IAC+FOjYuHX4AFYIVHQMTMR4g3DEYo60tBMaYsa01ACuOAz4hTkA3HKwE3RLQ45icwAM22hyYxM8tFgADvTDx2LAwRYAbhAwAO6IYmAHKoQehiAAUAEoXm8QuZHK4lo40GIsKDoTCQgByDEUNHo8zANq4vHIIl4iwojEZJbEYj0bEvMT0ahQCDUADWfwhxykdloB0YjgA1IKxG0xBiAJJoahw1bENgY8E4pFkvEo0mqzHYjWa-E63UhegZGBgWhwpZ3YwYRh-AHS6pI0HvNDZGCQ14qg1esRxOJiAASMDh81uYggcDEaCWUCg+u9ITGtFg+CgtDUoMlSKNJrNMAtVpt4ooYmdrvwCyw4POcfj5l9YgAQtSw8QMRH89aYMZJhA1CQxBgwAKa7WXMQ+BAcHziKCuRoeYnk6n0xiAIJiDLhAdDoOMtDG03my2dv700vGGDl25KsQARhOJyrI91hM9tZh9A7hbtQJBTpdF7us+BqLpey4ZlKYhfjAZ4AZeFZPm+74WC0wEwq+yHmPshzKrWSo1gAurheKISEr4tNWaBYPgWB8gwoJTDMxY3PcjwMFWZQVDAv5oLUIAAEz8Yg-EAKytO0ICYDgeD4NQcD9DQ9CMMwdStERIAsmgbK1KgkmdHgWDhIQjjQGUCLkHUJDEHccCIL6Bx3GyaiybccSGcQxnQAAAvx+C+fxblGSZUA0RAaC9ApxA5HcXTlDKEB3CMLQES0QA)

## Steps to Reproduce
1. Click on the button to increment the counter

## Expected Behavior
I expect the value of `vnode.dom` to be the actual dom in `onbeforeupdate`, as it is in `onupdate`. If this is not the expected behavior, maybe it should be explicited in the documentation.

## Current Behavior
Currently `vnode.dom` is undefined in `onbeforeupdate`

**Comment by osban** [comment]
Hello :) I think `vnode.dom` will always be `undefined` in `onbeforeupdate`. The reason why it shows the `dom` after a while, is that you're using a reference to the `vnode`, which by then has updated. If you deconstruct (or copy) the vnode, it'll stay `undefined`: [flems](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvAHgEIARAeQGEAVAJoAFAKIACEmQB8AHTQtJUMVAxoA5gF4ZIJttnzCMDABN9Y82JY5iGMdUIYATnBjEtIAKp8AYgFoAHHpyFpbEEMSwUnzwxCwA9GERMPrxhiYpAEa0xgCeKXFZufmK+pQgLrDUYfQIiCAAbIj1IAC+FOjYuHX4AFYIVHQMTMR4g3DEYo60tBMaYsa01ACuOAz4hTkA3HKwE3RLQ45icwAM22hyYxM8tFgADvTDx2LAwRYAbhAwAO6IYmAHKoQehiAAUAEoXm8QuZHK4lo40GIsKDoTCQgByDEUNHo8zANq4vHIIl4iwojEZJbEYj0bEvMT0ahQCDUADWfwhxykdloB0YjgA1IKxG0xBiAJJoahw1bENgY8E4pFkvEo0mqzHYjWa-E63UhegZGBgWhwpZ3YwYRh-AHS6pI0HABZYFqQ14qg1esRxOJiAASMDh81uYggcDEaCWUCg+u9ITGtFg+CgtDUoMlSKNJrNMAtVpt4ooIaw4POcfj5l9YgAQtSw8QMRH89aYMZJhA1CQxBgwAKK5WXMQ+BAcHziKCuRoeYnk6n0xiAIJiDLhHt9oOMtDG03my2tv70l1KsQARhOJzLA91hM9lZh9BbhbtQJBoPeaGyMHd14Ns5gKZphmUpiE+MD0h+X74Me5z3niLS-jCt5weY+yHMqlZKhWAC6GF4led63i05ZoFg+BYHyDCglMMzFjc9yPAwZZlBUMCvmgtQgAAzCciAnK07QgJgOB4Pg1BwP0ND0IwzB1K0uEgCyaBsrUqBCZ0eBYOEhCONAZQIuQdQkMQdxwIgvoHHcbJqGJtxxFpxA6dAAACABM+Dua59nabpUDkRAaC9JJxA5HcXTlDKEB3CMLTYS0QA).

**Comment by kfule** [comment]
You can access current dom via old.dom of onbeforeupdate(vnode, old).
[flems](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvAHgEIARAeQGEAVAJoAFAKIACEmQB8AHTQtJUMVAxoA5gF4ZIJttnzCMDABN9Y82JY5iGMdUIYATnBjEtIAKp8AYgFoAHHpyFpbEEMSwUnzwxCwA9GERMPrxhiYpAEa0xgCeKXFZufmK+pQgLrDUYfQIiCAAbIj1IAC+FOjYuHX4AFYIVHQMTMR4g3DEYo60tBMaYsa01ACuOAz4hTkA3HKwE3RLQ45icwAM22hyYxM8tFgADvTDx2LAwRYAbhAwAO6IYmAHKoQehiAAUAEoXm8QuZHK4lo40GIsKDoTCQgByDEUNHo8zANq4vHIIl4iwojEZJbEYj0bEvMT0ahQCDUADWfwhxykdloB0YjgA1IKxG0xBiAJJoahw1bENgY8E4pFkvEo0mqzHYjWa-E63UhegZGBgWhwpZ3YwYRh-AHS6pI0HvNDZGAURlQYyQ14qg1+3loOC0WD4KC0NSgyVIo0ms0wC1Wm3i93B4z4BZYcHnfX+kIuYh8CA4PnEUFcjQ8sbBmCh8ORgCCYgy4TEGDAAsZaGNpvNlutMD+9OdrvTtyVYgAjCcTlmc37Cb7c+j6An+7bAQ6wcPjDBvXP-VWQ2GI1GxKvGEOXTvR5nzkuyS193iF-eLPtDsql0q5wBdT942dFwXFpszQLB8CwPkGFBKYZndG57keBgs1adoQEwHA8Hwag4H6Gh6EYZg6laP8QBZNA2VqVB0M6PAsHCQhHGgMoEXIOoSGIO44EQOI4gOO42TUbDbjiejiEY6AAAEACZ8Fk6TRIYpioAgiA0F6PDiByO4unKGUIDuEYWh-FogA)

**Comment by martinboulais** [comment]
Hi,
Thank you for your quick reply. I used a reference on purpose, to be sure that the node that is provided is the one that will be used after the update, which makes sense.
The fact that it is named `old` mislead me. If I am not wrong, `old` actually contain the `current` vnode, and the first argument is passed the `future` vnode, which indeed do not have a dom element yet.
Closing the issue, thanks!
